test/StpTests/TacticTests.cpp

```diff
@@ -0,0 +1,91 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+using namespace rtt::ai::stp;
+
+class MockTactic : public Tactic{
+public:
+    StpInfo calculateInfoForSkill(const StpInfo &info) noexcept override {
+        return StpInfo();
+    }
+
+    void onInitialize() noexcept override {}
+
+    void onUpdate(const Status &status) noexcept override {}
+
+    void onTerminate() noexcept override {}
+
+    MOCK_METHOD1(isTacticFailingMock, bool(const StpInfo&));
+
+    bool isTacticFailing(const StpInfo &info) noexcept override {
+        return isTacticFailingMock(info);
+    }
+
+    MOCK_METHOD1(shouldTacticResetMock, bool(const StpInfo&));
+
+    bool shouldTacticReset(const StpInfo &info) noexcept override {
+        return shouldTacticResetMock(info);
+    }
+
+    MOCK_METHOD0(isEndTacticMock, bool());
+
+    bool isEndTactic() noexcept override {
+        return isEndTacticMock();
+    }
+};
+
+//TODO: nice to have, but too much for this: a before for each test case; see junit for details
+TEST(TacticTests, nonEndTacticFinishedSuccessful){
+    MockTactic tactic;
+    StpInfo info;
+    info.setBall(rtt::world_new::view::BallView(nullptr));
+    info.setField(rtt::ai::world::Field());
+    info.setRobot(rtt::world_new::view::RobotView(nullptr));
+
+    EXPECT_CALL(tactic, isEndTacticMock()).WillOnce(testing::Return(false));
+    auto result = tactic.update(info);
+    ASSERT_EQ(rtt::ai::stp::Status::Success, result);
+}
+
+TEST(TacticTests, endTacticFailingCondition){
+    MockTactic tactic;
+    StpInfo info;
+    info.setBall(rtt::world_new::view::BallView(nullptr));
+    info.setField(rtt::ai::world::Field());
+
+    proto::WorldRobot robot_proto;
+    robot_proto.set_id(1);
+    robot_proto.mutable_pos()->set_x(0.1);
+    std::unordered_map<uint8_t, proto::RobotFeedback> updateMap;
+    auto robot = rtt::world_new::robot::Robot(updateMap, robot_proto);
+    info.setRobot(rtt::world_new::view::RobotView(&robot));
+
+    EXPECT_CALL(tactic, isEndTacticMock()).WillOnce(testing::Return(true));
+    EXPECT_CALL(tactic, isTacticFailingMock(testing::_)).WillOnce(testing::Return(true));
+    auto result = tactic.update(info);
+    ASSERT_EQ(rtt::ai::stp::Status::Failure, result);
+}
+
+TEST(TacticTests, isTacticRunningSuccessful){
+    MockTactic tactic;
+    StpInfo info;
+    info.setBall(rtt::world_new::view::BallView(nullptr));
+    info.setField(rtt::ai::world::Field());
+
+    proto::WorldRobot robot_proto;
+    robot_proto.set_id(1);
+    robot_proto.mutable_pos()->set_x(0.1);
+    std::unordered_map<uint8_t, proto::RobotFeedback> updateMap;
+    auto robot = rtt::world_new::robot::Robot(updateMap, robot_proto);
+    info.setRobot(rtt::world_new::view::RobotView(&robot));
+
+    EXPECT_CALL(tactic, isEndTacticMock()).WillOnce(testing::Return(true));
+    EXPECT_CALL(tactic, isTacticFailingMock(testing::_)).WillOnce(testing::Return(false));
+    auto result = tactic.update(info);
+    ASSERT_EQ(rtt::ai::stp::Status::Running, result);
```

Comment:
Same here, can you put a brief summary of what you test and how it's tested here?